app/src/main/java/com/fivegmag/a5gmsmediastreamhandler/MediaSessionHandlerAdapter.kt

```diff
@@ -305,4 +309,23 @@ class MediaSessionHandlerAdapter() {
         }
     }
 
+    fun getIPAddress(ipVer: Int): String? {
+        val interfaces = NetworkInterface.getNetworkInterfaces()
+        while (interfaces.hasMoreElements()) {
+            val networkInterface = interfaces.nextElement()
+            val addresses = networkInterface.inetAddresses
+            while (addresses.hasMoreElements()) {
+                val address = addresses.nextElement()
+                if (!address.isLoopbackAddress && address.isSiteLocalAddress) {
+                    if ((ipVer == 4 && address.hostAddress.contains("."))||
+                        (ipVer == 6 && address.hostAddress.contains(":"))
+                    ){
+                        return address.hostAddress.toString()
+                    }
+                }
+            }
```

Comment:
Does this return the **server** IP address, as required or the client IP address?

When there is more than one interface on the UE does it just pick whichever is listed first?

Note that _ConsumptionReportingUnit.EndpointAddress_ is required to be populated with the address of the 5GMS AS (reference point M4), not that of the 5GMS AF (reference point M5). How does this function find the correct interface when the 5GMS AS and the 5GMS AF are reached via different PDU Sessions with different IP addresses?

Reply:
Yes, it returns the client's IP. If there is more than one interface on the UE, it will just pick the first.
Are you sure it should be the IP of server? Why can't I found it in the spec?
If  so, there seems to some issues when AF & AS are with different IP.

Reply:
I have realised that the definition of _mediaEndpointAddress_ is TS 26.512 clause 11.3.3.2 is ambiguous and could be interpreted as either the client endpoint or the server endpoint at reference point M4.

> The IP address and port number of the endpoint used to access the media consumed.

I have raised 5G-MAG/Standards#91 to try and get clarity from SA4 in the next few weeks.